File: programmers/level2/멀쩡한 사각형.py
import math

def solution(w,h):
    def get_gcd(a, b):
        if b==0: return a
        return get_gcd(b, a%b)
    
    if w == h: return w*h - w
    elif w == 1 or h == 1: return 0
    
    gcd = get_gcd(w, h) if w>=h else get_gcd(h, w)
    mini_w, mini_h = w//gcd, h//gcd
    leaning = mini_h/mini_w
    cnt = 0
    for x in range(mini_w):
        cnt += math.ceil(leaning*(x+1)) - math.floor(leaning*x)
    return w * h - cnt * gcd

File: programmers/level2/멀쩡한 사각형.java
class Solution {
    public long solution(int w, int h) {
        int gcd = getGcd(Math.max(w,h), Math.min(w,h));
        int miniW = w/gcd;
        int miniH = h/gcd;
        double leaning = (double)miniH/miniW;
        long miniAns = 0;
        for(int i=1; i<=miniW; i++) {
            double left = leaning * (i-1);
            double right = leaning * i;
            long leftH = (long)left;
            long rightH = (long)Math.ceil(right);
            miniAns+=rightH-leftH;
        }
        
        return (long)w*h - miniAns*gcd;
    }
    
    public int getGcd(int w, int h) {
        if(h==0) return w;
        return getGcd(h, w%h);
    }
}